### PR TITLE
GH Actions: lint & unit tests

### PR DESCRIPTION
**Pull request by dulmandakh**
Summary:
---------

Migrate Lint and Unit Test jobs to GitHub Actions. Will try to migrate e2e tests once merged.

Test Plan:
----------

Lint and Unit Test jobs will run on GitHub Actions, not Circle CI.

### CODE REVIEW

**Comment by dulmandakh** [comment]
@thymikee please review and merge

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -3,44 +3,62 @@ name: Node.js CI
...
 jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js 12
+        uses: actions/setup-node@v2
+        with:
+          node-version: 12
```
Comment (thymikee) [inline .github/workflows/test.yml:14]: let's use latest for linter

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -3,44 +3,62 @@ name: Node.js CI
...
+        run: yarn --frozen-lockfile
+      - name: Unit tests
+        run: yarn test:ci:unit
 
+  e2e-tests:
+    name: E2E Tests
     strategy:
       matrix:
-        node-version: [12.x]
+        node-version: [12]
```
Comment (thymikee) [inline .github/workflows/test.yml:43]: Let's use latest lts

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -3,44 +3,62 @@ name: Node.js CI
...
+          distribution: adopt
           java-version: 8
-          cache: 'gradle'
+          cache: gradle
 
       - name: Use Node.js ${{ matrix.node-version }}
         uses: actions/setup-node@v2
         with:
           node-version: ${{ matrix.node-version }}
+          cache: yarn
```
Comment (thymikee) [inline .github/workflows/test.yml:58]: it's that simple now? great!

Comment (dulmandakh) [reply .github/workflows/test.yml:58]: yep 👍

**Comment by dulmandakh** [comment]
@thymikee done 👍

**Review by thymikee — APPROVED** [review]
👍🏼

**Pull request merged**